index.js

```diff
@@ -195,6 +195,14 @@ function xgettext(input, options, cb) {
     if (options['files-from']) {
       input = fs.readFileSync(options['files-from'], options['from-code'])
         .split('\n')
+        .map(function (line) {
+          return line.split('').reduceRight(function(acc, c) {
+            if (' \t\r'.indexOf(c) === -1 || acc.length > 0) {
+              return c + acc;
+            }
+            return acc;
+          }, '');
```

Comment:
Also, having to loop through all characters just to remove the last ones feels quite heavy.
Can't this be done with a regex? Something like `line.replace(/[\t\r ]+$/, '')`?